akka-actor/src/main/scala/akka/io/DnsProvider.scala

```diff
@@ -6,8 +6,32 @@ package akka.io
 
 import akka.actor.Actor
 
+/**
+ * Where as it is possible to plug in alternative DNS implementations it is not recommended.
+ *
+ * It is expected that this will be deprecated/removed in future Akka versions
+ *
+ */
+// TODO make private and remove deprecated in 2.7.0
```

Comment:
issue link needed or coming soon to this PR?

Reply:
https://github.com/akka/akka/issues/28010